ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -90,6 +103,21 @@ def _get_measure_vals(self):
                     measure_val = np.mean(self.trainers[brain_name].reward_buffer)
                     brain_names_to_measure_vals[brain_name] = measure_val
             return brain_names_to_measure_vals
+
+        elif ((self.sampler_manager is not None) and (self.lesson_controller is not None)):
```

Comment:
Why do we need to check the sampler manager here? We don't use it.

Maybe it's just need, but I don't understand why the sampler needs to be this tied to to the lesson controller.

Reply:
The idea here is that we only use the computation in this section if we are doing generalization training, which is indicated by those two objects.